walnut/converter/converter.go

```diff
@@ -175,21 +189,57 @@ func GenerateTaskTemplate(extractedTasks []*task.Class, dumpFileName string) (er
 	return
 }
 
-func GenerateWorkflowTemplate(input workflow.Role, dumpFileName string) (err error) {
-	dir, _ := os.Getwd()
-	path := filepath.Join(dir, dumpFileName, "workflows")
+// confirmOverwrite takes an input string message to show the user and receive a yes/no response
+// if anything other than y/n provided, keep asking
+func confirmOverwrite (ask string) bool {
+	reader := bufio.NewReader(os.Stdin)
+
+	for {
+		fmt.Printf("%s [y/n]", ask)
+		response, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatal(err)
+		}
+		response = strings.ToLower(strings.TrimSpace(response))
+
+		if response == "y" || response == "yes" {
+			return true
+		} else if response == "n" || response == "no" {
+			return false
+		} else {
+			fmt.Printf("Invalid response\n")
+		}
+	}
+}
+
+func GenerateWorkflowTemplate(input workflow.Role, outputDir string) (err error) {
+	var dir string
+	if outputDir == "" {
+		dir, _ = os.Getwd()
+	} else {
+		dir = outputDir
+	}
+	path := filepath.Join(dir, "workflows")
+	path, _ = homedir.Expand(path)
 	os.MkdirAll(path, os.ModePerm)
 
 	yamlDATA, err := workflow.RoleToYAML(input)
 	if err != nil {
 		return fmt.Errorf("error converting role to YAML: %v", err)
 	}
 
-	fileName := filepath.Join(path, input.GetName() + ".yaml")
+	fileName := filepath.Join(path, input.GetName()+".yaml")
+	if _, err := os.Stat(fileName); os.IsNotExist(err) {} else {
```

Comment:
Constructs such as `if <condition> { //noop } else { //something }` are difficult to read and should be avoided in favor of `if !<condition> { //something }`.